integr8ly/keycloak-operator
# Add linkOnly and authenticateByDefault property to kc idp typ

## Motivation
This is required for creating a [Github](https://github.com/integr8ly/integreatly-operator/pull/61/files#diff-329c149b9583e9a4fcdb670a2ba866f0R316-R317) identity provider to be used for Launcher & Che in OpenShift 4.

## What
This adds the properties `linkOnly` to the [Keycloak Identity Provider](https://www.keycloak.org/docs-api/5.0/rest-api/index.html#_identityproviderrepresentation) type.

## Why
The Github identity provider created for Launcher and Che in OpenShift 4 requires the `Account Linking Only` to be set to true.

## Verification Steps
1. Run the operator
2. Create a Keycloak realm that has an identity provider with `LinkOnly` set to true.
  - **Verify**: Identity provider should have been created successfully
  - **Verify**: The property `Account Linking Only` of that identity provider should be set to true

## Checklist:


## Additional Notes
Related PR: https://github.com/integr8ly/integreatly-operator/pull/61
